receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +379,40 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) validateChannelHeader(ctx context.Context, resp http.ResponseWriter, req *http.Request) string {
+	// check channel header exists
+	partitionID := req.Header.Get(splunk.HTTPSplunkChannelHeader)
```

Comment:
one case i missed was users can technically send a payload without ack. but we are doing blanket validation here. If this is not included in the header, we should proceed without doing anything in ack.

We also missed a test case on: enabling ack extension and send a payload with channel header

Reply:
hmmm I'm not really following here, could you clarify?

Reply:
Scenario: User makes a call to ingest HEC data without channel header(no ack scenario)
Expected: user can ingest data without getting ack ID
Current behavior: user receives 401 validation error